### PR TITLE
ARC: forbid FIRQ or multiple register banks with 1 IRQ priority level

### LINKED ISSUE #38309: ARC context switch to interrupted thread busted with CONFIG_ARC_FIRQ=y and CONFIG_NUM_IRQ_PRIO_LEVELS=1
**Describe the bug**
The ARC can be configured with support for fast IRQs and only a single priority level.  In this configuration, fast IRQs is the only interrupt type supported by the hardware.

However, the `_switch_return_from_firq` implementation is expecting the lowest priority interrupt to perform a "regular" interrupt return and thus restore context from the stack.  This does not happen and the processor state is not restored correctly on such a context switch.  The system immediately goes off the rails as, among other things, neither the stack pointer nor the program counter are restored correctly.

**To Reproduce**
1. Configure the ARC interrupt controller with firq_option=true and number_of_levels=1, and configure Zephyr to match.
2. Run any application that performs a context switch to a previously-interrupted task (I called k_msleep twice: the first context switch to the idle thread works but the second crashes after the idle thread is interrupted by the timer).
3. Watch the processor do crazy things.

**Expected behavior**
Processor context should be correctly restored.

**Impact**
Behaviour is essentially random, processor typically faults after a few instructions and the system halts.

**Logs and console output**
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)

**Environment (please complete the following information):**
 - OS: Zephyr
 - Toolchain Zephyr SDK
 - Version 2.6

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
Don't allow to enable multiple register banks / fast interrupts if we have only one interrupt priority level. Such configurations are not supported in software and they are not beneficial from the performance point of view.

Fixes #39240
Fixes #38309

### CODE REVIEW

File: arch/arc/Kconfig [hunk 1]
```diff
@@ -145,17 +146,26 @@ config RGF_NUM_BANKS
...
 	depends on ISA_ARCV2
+	depends on NUM_IRQ_PRIO_LEVELS > 1
 	default y
 	help
 	  Fast interrupts are supported (FIRQ). If FIRQ enabled, for interrupts
 	  with highest priority, status32 and pc will be saved in aux regs,
 	  other regs will be saved according to the number of register bank;
 	  If FIRQ is disabled, the handle of interrupts with highest priority
 	  will be same with other interrupts.
+	  NOTE: we don't allow the configiration with FIRQ enabled and only one
```
Comment (ruuddw) [inline arch/arc/Kconfig:165]: typo, configuration

**Comment by ruuddw** [comment]
I think it makes sense to try and get it in 2.7, right? Can you remove draft, and add 2.7 label?

**Review by ruuddw — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

**Comment by cfriedt** [comment]
@evgeniy-paltsev - can you please link an issue to this? It is important as we are pretty far into our stabilization period. It will need to be triaged.

**Comment by evgeniy-paltsev** [comment]
@cfriedt I am wondering if we need to do something more in addition to linking issues to this PR?

**Comment by cfriedt** [comment]
@evgeniy-paltsev - right now we are only merging things into v2.7 that have the `Release Blocker` label. The release team was triaging items.

**Pull request merged**